Assets/Runtime/ScreenTransitions/SimpleFadeTransition.cs

```diff
@@ -20,27 +20,33 @@ public class SimpleFadeTransition : ATransitionComponent
         private Action currentAction;
         private Transform currentTarget;
 
-        private float startValue;
-        private float endValue;
+        private float startValue = 0f;
+        private float endValue = 0f;
 
         private bool shouldAnimate;
 
-        public override void Animate(Transform target, Action callWhenFinished) {
-            if (currentAction != null) {
-                canvasGroup.alpha = endValue;
-                currentAction();
-            }
-
+        public override void Animate(Transform target, Action callWhenFinished)
+        {
+            currentTarget = target;
             canvasGroup = target.GetComponent<CanvasGroup>();
-            if (canvasGroup == null) {
+            if (canvasGroup == null)
+            {
                 canvasGroup = target.gameObject.AddComponent<CanvasGroup>();
             }
 
-            if (fadeOut) {
+            if (currentAction != null)
```

Comment:
looks previous impl with calling `currentAction()` (actually it's previous action) before setting `canvasGroup` value correct, it works aka set final state for previous `canvasGroup`  if we call `Animate()` while previous animate not finished

what is the point to change it?

Reply:
what would be if we call this method for the first time without action and the canvas group is not exists there?

Reply:
not sure I get it, so pls correct me if I wrong/miss something - `Animate()` called by framework internally and guarantee that action will be always passed

Reply:
for now, we can guarantee that AUIScreenController would always pass OnTransitionInFinished/OnTransitionOutFinished actions, but, if there would be created some kind of additional implementation of AUIScreenController without Actions or for some reason canvasgroup would be destroyed - we would run into a problem here

Reply:
mb, so assume we can fix when (if) it happens

but it do not decline fact that right now we broke consistency - previously saved action could be called with new canvas group

Reply:
why there should be a new canvas group? we always push AUIScreenController transform as "target"

Reply:
yep, I agree that it should be on awake, can do that - if we are not going to call Animate() from screens that are not on the same gameobject